cmd/srcd/cmd/sql_test.go

```diff
@@ -1,28 +1,218 @@
+	command.Stdout = &out
+	command.Stderr = &out
+	command.Stdin = &in
+
+	io.WriteString(&in, "show tables;\n")
+	io.WriteString(&in, "describe table repositories;\n")
+
+	err := command.Start()
+	require.NoError(err)
+
+	err = command.Wait()
+	require.NoError(err)
+
+	require.Contains(out.String(), showTablesOutput)
+	require.Contains(out.String(), showRepoTableDescOutput)
+}
+
+func (s *SQLREPLTestSuite) TestInteractiveREPL() {
+	require := s.Require()
+
+	command, in, out, err := s.runInteractiveRepl()
+	require.NoError(err)
+
+	res := s.runInteractiveQuery(in, "show tables;\n", out)
+	require.Contains(res, showTablesOutput)
+
+	res = s.runInteractiveQuery(in, "describe table repositories;\n", out)
+	require.Contains(res, showRepoTableDescOutput)
+
+	require.NoError(s.exitInteractiveAndWait(10*time.Second, in, out))
+	require.NoError(s.waitMysqlCliContainerStopped(10 * time.Second))
+
+	command.Wait()
+}
+
+func (s *SQLREPLTestSuite) runInteractiveRepl() (*exec.Cmd, io.Writer, <-chan string, error) {
+	s.T().Helper()
+
+	command := s.CommandContext(context.TODO(), "sql")
+
+	ch := make(chan string)
+	cr := cmdtest.NewChannelWriter(ch)
+
+	command.Stdout = cr
+	command.Stderr = cr
+
+	in, err := pty.Start(command)
+	if err != nil {
+		panic(err)
+	}
+
+	linifier := cmdtest.NewStreamLinifier(1 * time.Second)
+	out := linifier.Linify(ch)
+	for s := range out {
+		if strings.HasPrefix(s, "mysql>") {
+			return command, in, out, nil
+		}
+	}
+
+	return nil, nil, nil, fmt.Errorf("Mysql cli prompt never started")
+}
+
+func (s *SQLREPLTestSuite) runInteractiveQuery(in io.Writer, query string, out <-chan string) string {
+	io.WriteString(in, query)
+
+	var res strings.Builder
+	for c := range out {
+		res.WriteString(c + "\r\n")
+		if s.containsSQLOutput(res.String()) {
+			break
+		}
+	}
+
+	return res.String()
+}
+
+func (s *SQLREPLTestSuite) exitInteractiveAndWait(timeout time.Duration, in io.Writer, out <-chan string) error {
+	io.WriteString(in, "exit;\n")
+
+	done := make(chan struct{})
+	go func() {
+		for c := range out {
+			if strings.Contains(c, "Bye") {
+				done <- struct{}{}
+				// don't return in order to consume all output and let the process exit
+			}
+		}
+	}()
+
+	select {
+	case <-done:
+		return nil
+	case <-time.After(timeout):
+		return fmt.Errorf("timeout of %v elapsed while waiting to exit", timeout)
+	}
+}
+
+func (s *SQLREPLTestSuite) waitMysqlCliContainerStopped(timeout time.Duration) error {
+	done := make(chan struct{})
+	go func() {
+		for {
+			running, err := docker.IsRunning(components.MysqlCli.Name, "")
+			if !running {
+				done <- struct{}{}
+				return
+			}
+
+			if err != nil {
+				panic(err)
+			}
+
+			time.Sleep(timeout / 5)
+		}
+	}()
+
+	select {
+	case <-done:
+		return nil
+	case <-time.After(timeout):
+		return fmt.Errorf("timeout of %v elapsed while waiting to stop container", timeout)
+	}
+}
+
+func (s *SQLREPLTestSuite) containsSQLOutput(out string) bool {
+	sep := regexp.MustCompile(`\+[-]+\+`)
```

Comment:
`-` inside `[]` has a special meaning not applied because there are no other chars surrounding it.
```suggestion
	sep := regexp.MustCompile(`\+-+\+`)
```
or
```suggestion
	sep := regexp.MustCompile(`\+[\-]+\+`)
```
but I don't see the benefit of `[]`

Reply:
Nice catch! I also just discovered that `-` has a special meaning only if is in-between two other chars, that why it was working.